patching/plugin-commands-patching/src/getPatchedDependency.ts

```diff
@@ -0,0 +1,56 @@
+import { searchForPackages, flattenSearchedPackages } from '@pnpm/list'
+import { parseWantedDependency, ParseWantedDependencyResult } from '@pnpm/parse-wanted-dependency'
+import { prompt } from 'enquirer'
+import { logger } from '@pnpm/logger'
+import type { Config } from '@pnpm/config'
+
+export async function getPatchedDependency ({
+  pkg,
+  selectedProjectsGraph,
+  lockfileDir,
+}: {
+  pkg: string
+  lockfileDir: string
+} & Pick<Config, 'selectedProjectsGraph'>): Promise<ParseWantedDependencyResult> {
+  const dep = parseWantedDependency(pkg)
+  let prefixes = [lockfileDir]
+  if (selectedProjectsGraph) {
+    prefixes = Object.values(selectedProjectsGraph).map(wsPkg => wsPkg.package.dir)
+  }
+  const pkgs = await searchForPackages([pkg], prefixes, {
+    depth: Infinity,
+    lockfileDir,
+  })
+
+  const versions = Array.from(new Set(flattenSearchedPackages(pkgs, { lockfileDir })
+    .map(({ version }) => version)))
+    .filter(Boolean) as string[]
+
+  if (dep.alias && dep.pref) {
+    if (!versions.length) {
+      logger.warn({
+        message: `Can not find ${dep.alias}@${dep.pref} in project ${lockfileDir}${versions.length ? `, you can specify currently installed version: ${versions.join(', ')} ` : ''}`,
+        prefix: lockfileDir,
+      })
```

Comment:
I also considered throwing an error and chose to show the warning for two reasons:

On the one hand, for projects that don't have executed `pnpm install`, print a warning would allow them to run the patch first and then install the dependencies during `patch-commit`.

On the other hand, for compatibility with the current behavior.

Also, I think it makes sense to report errors, do you think it's better to throw errors here?

Reply:
It is better to throw an error.

This error will happen also if there is no node_modules and no lockfile? In that case it is better to suggest running install in the error message.